core/src/main/scala/cats/data/Chain.scala

```diff
@@ -574,6 +574,71 @@ sealed abstract class Chain[+A] {
    */
   final def size: Long = length
 
+  /**
+   * The number of elements in this chain, if it can be cheaply computed, -1 otherwise.
+   * Cheaply usually means: Not requiring a collection traversal.
+   */
+  final def knownSize: Long =
+    this match {
+      case _ if isEmpty       => 0
+      case Chain.Singleton(_) => 1
+      case _                  => -1
+    }
+
+  /**
+   * Compares the length of this chain to a test value.
+   * 
+   * The method does not call `length` directly; its running time
+   * is `O(length min len)` instead of `O(length)`.
+   * 
+   * @param  len the test value that gets compared with the length.
+   * @return a negative value if `this.length < len`,
+   *         zero if `this.length == len` or
+   *         a positive value if `this.length > len`.
+   * @note   an adapted version of
+             [[https://github.com/scala/scala/blob/v2.13.8/src/library/scala/collection/Iterable.scala#L272-L288 Iterable#sizeCompare]]
+             from Scala Library v2.13.8 is used in a part of the implementation.
+   * 
+   * {{{
+   * scala> import cats.data.Chain
+   * scala> val chain = Chain(1, 2, 3)
+   * scala> val isLessThan4 = chain.lengthCompare(4) < 0
+   * scala> val isEqualTo3 = chain.lengthCompare(3) == 0
+   * scala> val isGreaterThan2 = chain.lengthCompare(2) > 0
+   * scala> isLessThan4 && isEqualTo3 && isGreaterThan2
+   * res0: Boolean = true
+   * }}}
+   */
+  final def lengthCompare(len: Long): Int = {
+    this match {
+      // `isEmpty` check should be faster than `== Chain.Empty`,
+      // but the compiler fails to prove that the match is still exhaustive.
+      case _ if isEmpty       => 0L.compareTo(len)
+      case Chain.Singleton(_) => 1L.compareTo(len)
```

Comment:
I think this call boxes. I think writing java.lang.Long.compare is safer since we can be 100% sure there is no boxing.

Reply:
Yes, it does. Really good catch, thank you. Fixed.